File: worker/Execution/NetworkScenarioParameters.cs
using System.Text.Json;

namespace Grdp.SoftwareIntegration.Worker.Execution;

internal static class NetworkScenarioParameters
{
    private static readonly HashSet<string> BoundaryFields = new(StringComparer.Ordinal)
    {
        "node", "pressure", "temperature", "flowRateType", "gasFlowRate", "liquidFlowRate", "massFlowRate"
    };

    internal static bool Valid(JsonElement value)
    {
        if (value.ValueKind == JsonValueKind.Null) return true;
        if (value.ValueKind != JsonValueKind.Object
            || value.EnumerateObject().Count() != 2
            || !value.TryGetProperty("schemaVersion", out var schema)
            || schema.ValueKind != JsonValueKind.String
            || schema.GetString() != "pipesim-network-parameters/1"
            || !value.TryGetProperty("boundaries", out var boundaries)
            || boundaries.ValueKind != JsonValueKind.Array
            || boundaries.GetArrayLength() is < 1 or > 64)
        {
            return false;
        }

        var nodes = new HashSet<string>(StringComparer.Ordinal);
        foreach (var boundary in boundaries.EnumerateArray())
        {
            if (boundary.ValueKind != JsonValueKind.Object
                || boundary.EnumerateObject().Any(item => !BoundaryFields.Contains(item.Name))
                || !boundary.TryGetProperty("node", out var node)
                || node.ValueKind != JsonValueKind.String)
            {
                return false;
            }

            var nodeName = node.GetString();
            if (string.IsNullOrWhiteSpace(nodeName) || nodeName.Length > 255
                || nodeName.Contains("..", StringComparison.Ordinal)
                || nodeName.IndexOfAny(['/', '\\', '\0']) >= 0
                || !nodes.Add(nodeName)) return false;

            foreach (var field in new[] { "pressure", "temperature", "gasFlowRate", "liquidFlowRate", "massFlowRate" })
            {
                if (!boundary.TryGetProperty(field, out var number)) continue;
                if (number.ValueKind != JsonValueKind.Number || !number.TryGetDouble(out var parsed)
                    || !double.IsFinite(parsed) || (field == "temperature" ? parsed <= -1000 : parsed <= 0)
                    || parsed > (field == "pressure" ? 100000 : field == "temperature" ? 100000 : 1000000000)) return false;
            }

            if (boundary.TryGetProperty("flowRateType", out var flowType))
            {
                if (flowType.ValueKind != JsonValueKind.String) return false;
                var type = flowType.GetString();
                var rateField = type switch
                {
                    "GasFlowRate" => "gasFlowRate",
                    "LiquidFlowRate" => "liquidFlowRate",
                    "MassFlowRate" => "massFlowRate",
                    _ => null
                };
                if (rateField is null || !boundary.TryGetProperty(rateField, out _)) return false;
            }
        }

        return true;
    }
}

File: backend/src/main/java/com/grdp/studio/softwareintegration/support/NetworkScenarioParameters.java
package com.grdp.studio.softwareintegration.support;

import tools.jackson.databind.JsonNode;

import java.util.HashSet;
import java.util.Set;

public final class NetworkScenarioParameters {
    private static final Set<String> BOUNDARY_FIELDS = Set.of(
            "node", "pressure", "temperature", "flowRateType", "gasFlowRate", "liquidFlowRate", "massFlowRate");

    private NetworkScenarioParameters() {}

    public static boolean valid(JsonNode value) {
        if (value == null || value.isNull()) return true;
        if (!value.isObject() || value.size() != 2
                || !"pipesim-network-parameters/1".equals(value.path("schemaVersion").asText())
                || !value.path("boundaries").isArray()
                || value.path("boundaries").size() < 1 || value.path("boundaries").size() > 64) return false;
        Set<String> nodes = new HashSet<>();
        for (JsonNode boundary : value.path("boundaries")) {
            if (!boundary.isObject() || boundary.propertyNames().isEmpty()) return false;
            for (String field : boundary.propertyNames()) if (!BOUNDARY_FIELDS.contains(field)) return false;
            String node = boundary.path("node").isTextual() ? boundary.path("node").asText() : null;
            if (node == null || node.isBlank() || node.length() > 255 || node.contains("..")
                    || node.indexOf('/') >= 0 || node.indexOf('\\') >= 0 || !nodes.add(node)) return false;
            if (!number(boundary, "pressure", 0, 100000)
                    || !number(boundary, "temperature", -1000, 100000)
                    || !number(boundary, "gasFlowRate", 0, 1000000000)
                    || !number(boundary, "liquidFlowRate", 0, 1000000000)
                    || !number(boundary, "massFlowRate", 0, 1000000000)) return false;
            if (boundary.has("flowRateType")) {
                String type = boundary.path("flowRateType").isTextual() ? boundary.path("flowRateType").asText() : null;
                String rateField = switch (type == null ? "" : type) {
                    case "GasFlowRate" -> "gasFlowRate";
                    case "LiquidFlowRate" -> "liquidFlowRate";
                    case "MassFlowRate" -> "massFlowRate";
                    default -> null;
                };
                if (rateField == null || !boundary.has(rateField)) return false;
            }
        }
        return true;
    }

    private static boolean number(JsonNode object, String field, double lowerExclusive, double upperInclusive) {
        if (!object.has(field)) return true;
        JsonNode value = object.get(field);
        double number = value.isNumber() ? value.asDouble() : Double.NaN;
        return Double.isFinite(number) && number > lowerExclusive && number <= upperInclusive;
    }
}
